src/Codeception/Suite.php

```diff
@@ -4,21 +4,288 @@
+            Notification::warning($e->getMessage(), '');
+            $warning = true;
+        } catch (Exception $e) {
+            $error = true;
+        } catch (Throwable $e) {
+            $e     = new ExceptionWrapper($e);
+            $error = true;
+        }
+
+        $time = $timer->stop()->asSeconds();
+        $test->addToAssertionCount(Assert::getCount());
+
+        if ($collectCodeCoverage) {
+            $append           = !$risky && !$incomplete && !$skipped;
+            $linesToBeCovered = [];
+            $linesToBeUsed    = [];
+
+            if ($append) {
+                try {
+                    $linesToBeCovered = (new CodeCoverageMetadataApi)->linesToBeCovered(
+                        $test::class,
+                        $test->getName(false)
+                    );
+
+                    $linesToBeUsed = (new CodeCoverageMetadataApi)->linesToBeUsed(
+                        $test::class,
+                        $test->getName(false)
+                    );
+                } catch (InvalidCoversTargetException $cce) {
+                    $result->addWarning(
+                        $test,
+                        new Warning(
+                            $cce->getMessage()
+                        ),
+                        $time
+                    );
+                }
+            }
+
+            try {
+                CodeCoverage::stop(
+                    $append,
+                    $linesToBeCovered,
+                    $linesToBeUsed
+                );
+            } catch (UnintentionallyCoveredCodeException $cce) {
+                $unintentionallyCoveredCodeError = new RiskyDueToUnintentionallyCoveredCodeException(
+                    'This test executed code that is not listed as code to be covered or used:' .
+                    PHP_EOL . $cce->getMessage()
+                );
+            } catch (OriginalCodeCoverageException $cce) {
+                $error = true;
+
+                $e = $e ?? $cce;
+            }
+        }
+
+        if ($error && isset($e)) {
+            $result->addError($test, $e, $time);
+            $eventType = Events::TEST_ERROR;
+        } elseif ($failure && isset($e)) {
+            $result->addFailure($test, $e, $time);
+            if ($skipped) {
+                $eventType = Events::TEST_SKIPPED;
+            } elseif($incomplete) {
+                $eventType = Events::TEST_INCOMPLETE;
+            } else {
+                $eventType = Events::TEST_FAIL;
+            }
+        } elseif ($warning && isset($e)) {
+            $result->addWarning($test, $e, $time);
+            $eventType = Events::TEST_WARNING;
+        } elseif (isset($unintentionallyCoveredCodeError)) {
+            $result->addFailure(
+                $test,
+                $unintentionallyCoveredCodeError,
+                $time
+            );
+            $eventType = Events::TEST_ERROR;
+        }
+//        elseif ($this->configuration->reportUselessTests() &&
```

Comment:
because we don't have risky or useless tests, right?
maybe this code should be removed

Reply:
I promised to @TavoNiievez to implement risky tests in Codeception codebase.